AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginLazyLoadTests/LL1/AWSDataStoreLazyLoadPostComment4V2StressTests.swift

```diff
@@ -0,0 +1,211 @@
+        /// Maintain the latest state of whether the outbox is empty or not in `isOutboxEmpty` variable.
+        /// Fulfill `savesSyncedExpectation` after all tasks have been created and the outbox is empty.
+        private func subscribeToOutboxEvent(_ savesSyncedExpectation: XCTestExpectation) {
+            self.subscribeToOutboxEventTask = Task {
+                for await event in Amplify.Hub.publisher(for: .dataStore).values {
+                    switch event.eventName {
+                    case HubPayload.EventName.DataStore.outboxStatus:
+                        guard let outboxEvent = event.data as? OutboxStatusEvent else {
+                            return
+                        }
+                        isOutboxEmpty = outboxEvent.isEmpty
+                        outboxEventsCount += 1
+                        Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] \(outboxEventsCount) isOutboxEmpty: \(isOutboxEmpty), index: \(index)")
+                        if index == (count - 1)  && isOutboxEmpty {
+                            XCTAssertEqual(savedModels.count, count)
+                            savesSyncedExpectation.fulfill()
+                        }
+                    default:
+                        break
+                    }
+                }
+            }
+        }
+    }
+
+    func assertQueryComment(_ savedComment: Comment, post: Post) async throws {
+        guard let persistedComment = try await Amplify.DataStore.query(
+            Comment.self,
+            byIdentifier: savedComment.identifier) else {
+            Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] Skipping comment \(savedComment.id) since it is not persisted in local DB")
+            return
+        }
+
+        let result = try await Amplify.API.query(
+            request: .get(
+                Comment.self,
+                byIdentifier: savedComment.id))
+        switch result {
+        case .success(let comment):
+            guard let comment else {
+                XCTFail("Missing comment, should contain \(savedComment)")
+                return
+            }
+            assertLazyReference(
+                comment._post,
+                state: .notLoaded(
+                    identifiers: [.init(
+                        name: "id",
+                        value: post.identifier)]))
+        case .failure(let error):
+            XCTFail("Failed to query, error \(error)")
+        }
+    }
+
+    func assertQueryPost(_ savedPost: Post) async throws {
+        guard let persistedPost = try await Amplify.DataStore.query(
+            Post.self,
+            byIdentifier: savedPost.identifier) else {
+            Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] Skipping post \(savedPost.id) since it is not persisted in local DB")
+            return
```

Comment:
i think we can add a check here that if `persistedPost` is not in DB, then it should not have synced as well through `API.query()` call

Reply:
updated